libsolidity/formal/BMC.cpp

```diff
@@ -701,6 +708,55 @@ void BMC::visitAddMulMod(FunctionCall const& _funCall)
 	SMTEncoder::visitAddMulMod(_funCall);
 }
 
+void BMC::visitTypeConversion(FunctionCall const& _funCall)
+{
+	auto argument = _funCall.arguments().front();
+	auto const argType = argument->annotation().type;
+	auto const funCallType = _funCall.annotation().type;
+	auto const expressionType = _funCall.expression().annotation().type;
+	auto const resultType = dynamic_cast<TypeType const&>(*expressionType).actualType();
+
+	auto symbArg = expr(*argument, funCallType);
+
+	// do not warn about address(contract)
+	if (smt::isAddress(*resultType) && dynamic_cast<ContractType const*>(argType))
+		return;
+
+	if (smt::isStringLiteral(*argType) && smt::isFixedBytes(*resultType))
+		return;
+
+	ArrayType const* arrayType = dynamic_cast<ArrayType const*>(argType);
+
+	// no support for array slice
+	if (dynamic_cast<ArraySliceType const*>(argType))
+		return;
+
+	if (arrayType && arrayType->isByteArrayOrString() && smt::isFixedBytes(*resultType))
```

Comment:
We do not need `arrayType` outside of the corresponding `if` block, no?

```suggestion
	// no support for array slice
	if (dynamic_cast<ArraySliceType const*>(argType))
		return;

	if (ArrayType const* arrayType = dynamic_cast<ArrayType const*>(argType); arrayType && arrayType->isByteArrayOrString() && smt::isFixedBytes(*resultType))
```